### PR TITLE
Opensearch module refactor

### PR DESCRIPTION
**Pull request by Arunsai14**
## Description

 Terraform module to deploy and configure an Amazon OpenSearch Service domain with customizable settings from scratch

## Related Issue

- [Issue Number](https://github.com/sourcefuse/terraform-aws-arc-opensearch/issues/28): Brief description

## Checklist

Please ensure that the following steps are completed before submitting the pull request:

- [x] Code follows the Terraform best practices and style guidelines.
- [x] Changes are appropriately documented, including any necessary updates to README or other documentation files.
- [x] Unit tests have been added or updated to cover the changes introduced by this pull request.
- [x] Changes have been tested locally and verified to work as expected.
- [x] The code has been reviewed to ensure it aligns with the project's goals and standards.
- [x] Dependencies and backward compatibility have been considered and addressed if applicable.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Testing Instructions
Ran `terraform apply` in Example Folder
image

## Screenshots

image

## Additional Notes

Add any additional notes or context that might be helpful for reviewers or users testing the changes.

### CODE REVIEW

**Comment by vijay-stephen** [comment]
@Arunsai14 rename non-vpc to public

File: variables.tf [hunk 1]
```diff
@@ -1,226 +1,405 @@
-################################################################################
-## shared
-################################################################################
-variable "environment" {
+variable "region" {
+  description = "AWS region"
   type        = string
-  description = "Name of the environment, i.e. dev, stage, prod"
+  default     = "us-east-1"
 }
```
Comment (vijay-stephen) [inline variables.tf:5]: Do we need region variable

File: variables.tf [hunk 2]
```diff
@@ -1,226 +1,405 @@
...
 
-variable "name" {
+variable "domain_name" {
+  description = "Name of the OpenSearch domain"
   type        = string
-  description = "Name of the OpenSearch resource"
 }
```
Comment (vijay-stephen) [inline variables.tf:10]: rename it as `name`

File: variables.tf [hunk 3]
```diff
@@ -1,226 +1,405 @@
...
+  default     = "OpenSearch_1.0"
 }
 
-variable "tags" {
-  type        = map(string)
-  description = "Default tags to apply to every resource"
+variable "instance_type" {
+  description = "Instance type for the OpenSearch domain"
+  type        = string
+  default     = "m4.large.search"
```
Comment (vijay-stephen) [inline variables.tf:21]: Please use least type

File: variables.tf [hunk 4]
```diff
@@ -1,226 +1,405 @@
...
+}
+
+variable "instance_count" {
+  description = "Number of instances in the cluster"
+  type        = number
+  default     = 2
```
Comment (vijay-stephen) [inline variables.tf:27]: Please use 1

File: variables.tf [hunk 5]
```diff
@@ -1,226 +1,405 @@
...
 }
-variable "security_group_ids" {
-  description = "List of security groups to assign OpenSearch"
-  type        = list(string)
-  default     = []
+
+variable "kms_key_id" {
+  description = "KMS key ID for encryption at rest"
+  type        = string
+  default     = ""
```
Comment (vijay-stephen) [inline variables.tf:135]: default it as null

File: variables.tf [hunk 6]
```diff
@@ -1,226 +1,405 @@
...
-  default     = []
-}
-variable "anonymous_iam_actions" {
-  type        = list(string)
-  description = "List of actions to allow for the anonymous (`*`) IAM roles, _e.g._ `es:ESHttpGet`, `es:ESHttpPut`, `es:ESHttpPost`"
-  default     = []
+variable "tls_security_policy" {
+  description = "TLS security policy for HTTPS endpoints"
+  type        = string
+  default     = "Policy-Min-TLS-1-2-2019-07"
```
Comment (vijay-stephen) [inline variables.tf:153]: please use latest policy

File: variables.tf [hunk 7]
```diff
@@ -1,226 +1,405 @@
...
-  description = "List of availability zones to deploy the cluster in."
-  default = [
-    "us-east-1a",
-    "us-east-1b"
-  ]
+variable "enable_vpc_options" {
+  description = "Enable VPC options for the OpenSearch domain."
+  type        = bool
+  default     = false # Set a default value or leave it out if it's required
+}
```
Comment (vijay-stephen) [inline variables.tf:338]: please use vpc_id variable, check if its value is null

**Review by vijay-stephen — CHANGES_REQUESTED** [review]

**Comment by vijay-stephen** [comment]
@Arunsai14 Add serverless option as well

File: variables.tf (hunk 1 above)
Comment (Arunsai14) [reply variables.tf]: correction made

File: variables.tf (hunk 3 above)
Comment (Arunsai14) [reply variables.tf]: correction made

**Review by manikanta-sadurla — APPROVED** [review]

**Review by siddharth-singh1 — APPROVED** [review]

**Review by tsaucier-sf — APPROVED** [review]

**Review by vijay-stephen — APPROVED** [review]

**Pull request merged**